crypto/ml_dsa/ml_dsa_vector.h

```diff
@@ -0,0 +1,257 @@
+{
+    size_t i;
+    uint8_t derived_seed[ML_DSA_RHO_PRIME_BYTES + 2];
+
+    memcpy(derived_seed, rho_prime, ML_DSA_RHO_PRIME_BYTES);
+
+    for (i = 0; i < out->num_poly; i++) {
+        size_t index = kappa + i;
+
+        derived_seed[ML_DSA_RHO_PRIME_BYTES] = index & 0xFF;
+        derived_seed[ML_DSA_RHO_PRIME_BYTES + 1] = (index >> 8) & 0xFF;
+        poly_expand_mask(out->poly + i, derived_seed, sizeof(derived_seed),
+                         gamma1, h_ctx);
+    }
+}
+
+/* Scale back previously rounded value */
+static ossl_inline ossl_unused void
+vector_scale_power2_round_ntt(const VECTOR *in, VECTOR *out)
+{
+    size_t i;
+
+    for (i = 0; i < in->num_poly; i++)
+        poly_scale_power2_round(in->poly + i, out->poly + i);
+    vector_ntt(out);
+}
+
+/*
+ * @brief Decompose all polynomial coefficients of a vector into (t1, t0) such
+ * that coeff[i] == t1[i] * 2^13 + t0[i] mod q.
+ * See FIPS 204, Algorithm 35, Power2Round()
+ */
+static ossl_inline ossl_unused void
+vector_power2_round(const VECTOR *t, VECTOR *t1, VECTOR *t0)
+{
+    size_t i;
+
+    for (i = 0; i < t->num_poly; i++)
+        poly_power2_round(t->poly + i, t1->poly + i, t0->poly + i);
+}
+
+static ossl_inline ossl_unused void
+vector_high_bits(const VECTOR *in, uint32_t gamma2, VECTOR *out)
+{
+    size_t i;
+
+    for (i = 0; i < out->num_poly; i++)
+        poly_high_bits(in->poly + i, gamma2, out->poly + i);
+}
+
+static ossl_inline ossl_unused void
+vector_low_bits(const VECTOR *in, uint32_t gamma2, VECTOR *out)
+{
+    size_t i;
+
+    for (i = 0; i < out->num_poly; i++)
+        poly_low_bits(in->poly + i, gamma2, out->poly + i);
+}
+
+static ossl_inline ossl_unused uint32_t
+vector_max(const VECTOR *v)
+{
+    size_t i;
+    uint32_t mx = 0;
+
+    for (i = 0; i < v->num_poly; i++)
+        poly_max(v->poly + i, &mx);
+    return mx;
+}
+
+static ossl_inline ossl_unused uint32_t
+vector_max_signed(const VECTOR *v)
+{
+    size_t i;
+    uint32_t mx = 0;
+
+    for (i = 0; i < v->num_poly; i++)
+        poly_max_signed(v->poly + i, &mx);
+    return mx;
+}
+
+static ossl_inline ossl_unused size_t
+vector_count_ones(const VECTOR *v)
+{
+    int j;
+    size_t i, count = 0;
+
+    for (i = 0; i < v->num_poly; i++)
+        for (j = 0; j < ML_DSA_NUM_POLY_COEFFICIENTS; j++)
+            count += v->poly[i].coeff[j];
+    return count;
+}
+
+static ossl_inline ossl_unused void
+vector_make_hint(const VECTOR *ct0, const VECTOR *cs2, const VECTOR *w,
+                 uint32_t gamma2, VECTOR *out)
+{
+    size_t i;
+
+    for (i = 0; i < out->num_poly; i++)
+        poly_make_hint(ct0->poly + i, cs2->poly + i, w->poly + i, gamma2,
+                       out->poly + i);
+}
+
+static ossl_inline ossl_unused void
+vector_use_hint(const VECTOR *h, const VECTOR *r, uint32_t gamma2, VECTOR *out)
+{
+    size_t i;
+
+    for (i = 0; i < out->num_poly; i++)
+        poly_use_hint(h->poly + i, r->poly + i, gamma2, out->poly + i);
+}
+
+#if defined(ML_DSA_DEBUG)
```

Comment:
So you really do mean debug - i.e. functions that are otherwise unused for calling when in the debugger ... are users ever  going to want a vector print out in text format?

Reply:
They were real useful when debugging and trying to figure out where something went wrong.

Reply:
removed from poly also